### PR TITLE
Switch to using eslint

### PR DESCRIPTION
**Pull request by bryceosterhaus**
Dependent on https://github.com/metal/gulp-metal/pull/25

### CODE REVIEW

File: packages/metal-state/test/validators.js [hunk 1]
```diff
@@ -211,10 +211,12 @@ describe('validators', function() {
 		const NAME = 'name';
 		const PARENT_COMPONENT_NAME = 'parentComponent';
 
+		/* eslint-disable */
 		const ERROR_MESSAGE = `Error: Warning: Invalid state passed to '${NAME}'. ` +
```
Comment (mairatma) [inline packages/metal-state/test/validators.js:214]: Just curious, why does eslint complain about this? Is it an anti pattern to use the new string templates with concatenation?

**Comment by bryceosterhaus** [comment]
Just updated to include `{ "allowTemplateLiterals": true }` for strings

File: gulpfile.js [hunk 2]
```diff
@@ -21,8 +22,7 @@ metal.registerTasks({
 	bundleFileName: 'metal.js',
 	formatGlobs: codeGlobs,
 	karma: require('karma'),
-	// TODO: Find a way to lint jsx files (maybe use eslint instead of jshint).
-	lintGlobs: codeGlobs.concat('!packages/metal-jsx/test/**/*.js'),
```
Comment (mairatma) [inline gulpfile.js:25]: I'm so glad we can remove that now :)

**Review by mairatma — COMMENTED** [review]
I just thought of something that might be useful while we're still in this transition phase between jshint and eslint. What if we add an option to **gulp-metal**, something like `useEslint`, that when set to `true` will have `gulp lint` use eslint instead of jshint? That way we don't need to know which task to use for each repo, we can just set this for all that were converted to eslint and keep using `gulp lint` on all of them. Could you do that change on **gulp-metal** as well, @bryceosterhaus? Once that's in we could publish it and merge this too.

**Comment by bryceosterhaus** [comment]
@mairatma  updated pr.

Also see https://github.com/metal/gulp-metal/pull/28

**Comment by mairatma** [comment]
Perfect, let's just wait for **gulp-metal** to be released with that change. I'll do it today, after the complexity [pr](https://github.com/metal/gulp-metal/pull/27) is in as well.

**Comment by mairatma** [comment]
@bryceosterhaus, I was about to merge this but then I realized that the commit names are not in the expected format yet. Could you rename them to that they follow the usual pattern (that is, assume the phrase starts with `It`, as in `Adds eslintrc and remove jshintrc`)?

**Comment by bryceosterhaus** [comment]
@mairatma just pushed changes to commit names. Let me know if those are better.

Thanks!

**Comment by mairatma** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mairatma** [comment]
Thanks, pr merged :)

**Pull request closed without merging**